documents/coredocument.go

```diff
@@ -651,14 +651,24 @@ func (cd *CoreDocument) Timestamp() (time.Time, error) {
 }
 
 // AddAttributes adds a custom attribute to the model with the given value. If an attribute with the given name already exists, it's updated.
-func (cd *CoreDocument) AddAttributes(attrs ...Attribute) (*CoreDocument, error) {
+func (cd *CoreDocument) AddAttributes(ca *CollaboratorsAccess, prepareNewVersion bool, attrs ...Attribute) (*CoreDocument, error) {
 	if len(attrs) < 1 {
 		return nil, errors.NewTypedError(ErrCDAttribute, errors.New("require at least one attribute"))
 	}
 
-	ncd, err := cd.PrepareNewVersion(nil, CollaboratorsAccess{}, nil)
-	if err != nil {
-		return nil, errors.NewTypedError(ErrCDAttribute, errors.New("failed to prepare new version: %v", err))
+	if ca == nil {
```

Comment:
if `ca` only has meaning for a prepareNewVersion context, then lets move this init check there.

Reply:
resolved